src/odbc_connection.cpp

```diff
@@ -1249,8 +1249,56 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
 
       SQLRETURN return_code;
 
+      // allocate a new statement handle
+      uv_mutex_lock(&ODBC::g_odbcMutex);
+
+      return_code =
+      SQLAllocHandle
+      (
+        SQL_HANDLE_STMT,            // HandleType
+        odbcConnectionObject->hDBC, // InputHandle
+        &data->hstmt                // OutputHandlePtr
+      );
+      uv_mutex_unlock(&ODBC::g_odbcMutex);
+      if (!SQL_SUCCEEDED(return_code)) {
+        this->errors = GetODBCErrors(SQL_HANDLE_DBC, odbcConnectionObject->hDBC);
+        SetError("[odbc] Error allocating a statment handle to get procedure information\0");
+        return;
+      }
+
+      // Create the statement to call the stored procedure using the ODBC Call
+      // escape sequence.
+
+      // Create the string containing the parameter markers "?,?,?,?" where the
+      // number of '?' is the number of parameters. Multiply the number of
+      // parameters by 2 to account for commas and a \0 terminator.
+
+      size_t parameterStringSize = (data->parameterCount * 2);
+      char *parameterString = new char[parameterStringSize];
+      parameterString[0] = '\0';
+
+      for (int i = 0; i < data->parameterCount; i++) {
+        if (i == (data->parameterCount - 1)) {
+          strcat(parameterString, "?"); // for last parameter, don't add ','
+        } else {
+          strcat(parameterString, "?,");
+        }
+      }
+
+      SQLUINTEGER procedure_name_length = 0;
+      if (data->catalog)
+      {
+        procedure_name_length += strlen((const char *)data->catalog) + 1;
+      }
+      if (data->schema)
+      {
+        procedure_name_length += strlen((const char *)data->schema) + 1;
+      }
+      // "procedure" is always passed to this API
+      procedure_name_length += strlen((const char *)data->procedure);
+
       #ifndef UNICODE
-      char *combinedProcedureName = new char[1024]();
+      char *combinedProcedureName = new char[procedure_name_length + 1]();
```

Comment:
A lot of this code could be simplified by using a std::string as well. Could just reserve say 256 or 512 bytes which should be enough in the majority of cases for the whole CALL statement and then just use std::string appends to put it all together (or std::wstring with UNICODE). In the event that not enough space was reserved, the string will automatically resize itself and continue on and then it will automatically free it's memory when it destructs too.

So code is simpler and no manual memory management.